src/wp-includes/template.php

```diff
@@ -698,20 +698,36 @@ function get_attachment_template() {
  */
 function locate_template( $template_names, $load = false, $require_once = true, $args = array() ) {
 	$located = '';
+
+	static $template_cache = array();
 	foreach ( (array) $template_names as $template_name ) {
 		if ( ! $template_name ) {
 			continue;
 		}
+
+		if ( isset( $template_cache[ $template_name ] ) && '' === $template_cache[ $template_name ] ) {
+			continue;
+		}
+
+		if ( isset( $template_cache[ $template_name ] ) ) {
+			$located = $template_cache[ $template_name ];
+			break;
+		}
```

Comment:
Just wondering if it's worth calling `isset( $template_cache[ $template_name ] )` twice here, rather than:

```php
if ( isset( $template_cache[ $template_name ] ) ) {
	if ( '' === $template_cache[ $template_name ] ) {
		continue;
	}

	$located = $template_cache[ $template_name ];
	break;
}
```
@mukeshpanchal27 What do you think? Would you prefer to keep the conditions separate, or to reduce the number of conditions?

Reply:
@costdev the change make sense to me.